### ISSUE TITLE
[Stability] Fable tests

### ISSUE THREAD

**Issue by delneg** [delneg/source-map-sharp#8]
I wanted to make some changes to the code, possibly benchmark it and overall 'break' it a little bit.
The tests for dotnet realisation are there, so I'm pretty confident that it won't break anybody's code who's using it, but then i remembered that Fable support was also contributed by @ncave .
Is it possible to get some code related to that in PR, maybe some tests that someone has or build script ?

cc @ncave @alfonsogarciacaro help would be appreciated :)

**Comment by alfonsogarciacaro** [comment]
Sorry for the late reply @delneg, I missed the notification. I see the tests use the `Fact` attribute, I guess we could turn this into a JS decorator that transforms the functions into Mocha tests.

Unfortunately I don't have much time now to look into this, I will try to send a PR after finishing other tasks. It's probably not too urgent anyways as I don't know of any Fable app using source-map-sharp (I think fable-compiler-js doesn't use it yet) so the dotnet tests should be enough to ensure the stability of the tool upon changes.

**Comment by delneg** [comment]
I agree that the test would be sufficient, but as of right now I don't even know how to run this package with Fable. 
I mean, do I just `dotnet fable  source-map-sharp.fsproj` and that's it ?
And how I actually convert test lol...

**Comment by delneg** [comment]
It seems that there was `package.json` that I didn't notice for some time..
Anyway, I managed to understand how it is and get it working in https://github.com/delneg/source-map-sharp/commit/b9353482baff0d6cafb16131bb7ff3523d060b2b
Though not with the same Test project but a separate one

P.S. Really got dragged off the tracks with Fable.Mocha 1.0.0 in it's README... Didn't know that actually there was 2.11.0 until like an hour or two

**Comment by delneg** [comment]
As of right now, Fable version seems pretty stable:
```
> test
> mocha build/tests/Tests.js



  Util
    √ test relative()

  Base64
    √ Base64 encode test out of range encoding
    √ Base64 encode test out of range encoding#2
    √ test normal encoding and decoding
    √ test normal encoding and decoding for VLQ

  ArraySet
    √ test .has() membership
    √ test .indexOf() elements
    √ test .at() indexing
    √ test creating from an array
    √ test that you can add __proto__; see github issue #30
    √ test .fromArray() with duplicates
    √ test .add() with duplicates
    √ test .size()
    √ test .size() with disallowed duplicates
    √ test .size() with allowed duplicates

  SourceGenerator
    √ test some simple stuff
    √ test JSON serialization 1
    √ test JSON serialization 2
    √ test adding mappings (case 1)
    √ test adding mappings (case 2)
    √ test adding mappings (case 3)
    √ test adding mappings (invalid)
    √ test adding mappings with skipValidation
    √ test that the correct mappings are being generated
    √ test that adding a mapping with an empty string name does not break generation
    √ test that source content can be set
    √ test sorting with duplicate generated mappings
    √ test ignore duplicate mappings.
    √ test github issue #72, check for duplicate names or sources
    √ test setting sourcesContent to null when already null
    √ test numeric names #231


  31 passing (33ms)

```

**Comment by delneg** [comment]
Closed via #10
